Fix COMPONENT_CLIMATE config

QRPp opened the pull request:
Including attr_t, cmd_t and stat_t variables breaks configuration of HA MQTT
climate platform.

pimvanpelt commented on src/mgos_homeassistant_api.c:
```diff
@@ -567,9 +567,11 @@ static bool mgos_homeassistant_object_send_config_mqtt(struct mgos_homeassistant
-  if (mgos_homeassistant_object_get_attr(o, NULL)) json_printf(&payload, ",attr_t:%Q", "~/attr");
+  if (o->component != COMPONENT_CLIMATE) {
+    json_printf(&payload, ",stat_t:%Q", "~");
+    if (mgos_homeassistant_object_get_cmd(o, NULL)) json_printf(&payload, ",cmd_t:%Q", "~/cmd");
```
For the cmd and attr, you should set the command to NULL, eg `mgos_homeassistant_object_add_cmd_cb(o, NULL, NULL);` and similar for `mgos_homeassistant_object_add_attr_cb(o, NULL, NULL);` and in the case of these callbacks being NULL, they won't be added to the payload here.

Are you sure `stat_t` is invalid? That would be unfortunate, because it is assumed to be mandatory. I do not approve of special casing component types here. But perhaps we can do better, by adding a topic bitmask of sorts at creation time, adding an enabler/disabler in the API and checking the mask here ?

QRPp replied:
Thanks for the reply.

First, per stat:
- Re `stat_t` aka `state_topic`.  The [MQTT HVAC](https://www.home-assistant.io/integrations/climate.mqtt) specification doesn't list it.  I wouldn't even know, had my HA installation not refused to accept a `climate` config with these variables (see error message [*] at the bottom of the message).
- Re `cmd_t`.  My setup uses `~/cmd` for `hold_cmd_t`, `mode_cmd_t` etc.  Using `~/cmd/foo` _could_ be a workaround (but see [**] at the bottom of the message), though that seems inferior when the error could be fixed instead.  Adding `cmd_t` to a `climate` config is plain wrong as per the [same spec](https://www.home-assistant.io/integrations/climate.mqtt), why should the library code do that?
- Re `attr_t`.  I didn't use it, however the same thinking as for `cmd_t` applies: a robust library had better not knowingly break a specification.

Now, re special-casing: good!  I dislike that sort of approach too.  However I thought I'd raise the matter before going all-in re a comprehensive fix.  Wasn't sure whether it'd go anywhere, after all, [another PR](https://github.com/mongoose-os-libs/homeassistant/pull/2) has been hanging for ~a month — you may be very busy...  So, I could cobble up a per-component filter lookup, shall I?

[*] The log excerpt where discovery slips on exactly the `state_topic`:
```
... ERROR (MainThread) [homeassistant.util.logging] Exception in discovery_callback when dispatching 'mqtt_discovery_updated_('climate', ...
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 418, in discovery_callback
    await self._discovery_update(payload)
  File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 561, in discovery_update
    config = self.config_schema()(discovery_payload)
  File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: extra keys not allowed @ data['state_topic']

```
[**] [This line](https://github.com/mongoose-os-libs/homeassistant/blob/master/src/mgos_homeassistant_api.c#L616) forced me to add `~/cmd` early on, otherwise the config wouldn't be emitted at all.  No, the setup didn't include a central status callback.  Yes, it's a weird climate control device.  In any case, conditioning sending config on `status_cb` rather than, e.g., `json_config_additional_payload` seemed dubious, however I may not understand the rationale.

QRPp committed:
Fix COMPONENT_CLIMATE config

Including attr_t, cmd_t and stat_t variables breaks configuration of HA MQTT
climate platform.
